File: concrete/src/main/java/pcb/uwu/unit/base/NauticalMileUnit.java
package pcb.uwu.unit.base;

import java.math.BigDecimal;

import static java.math.MathContext.DECIMAL64;

public class NauticalMileUnit extends LengthUnit {

	private static final BigDecimal METERS_PER_NAUTICAL_MILE = new BigDecimal("1852");

	public static final NauticalMileUnit NAUTICALMILE = new NauticalMileUnit();

	private NauticalMileUnit() {
		super("mile", "mile", "miles",
				value -> value.multipliedBy(METERS_PER_NAUTICAL_MILE, DECIMAL64),
				value -> value.dividedBy(METERS_PER_NAUTICAL_MILE, DECIMAL64));
	}
}

File: concrete/src/main/java/pcb/uwu/unit/base/NauticalMileUnit.kt
package pcb.uwu.unit.base

import java.math.BigDecimal

object NauticalMileUnit
    : LengthUnit(symbol = "mile",
                 singularName = "mile",
                 pluralName = "miles",
                 translationToCanonical = { it * BigDecimal("1852") },
                 translationFromCanonical = { it / BigDecimal("1852") })
